### PR TITLE
[inline model resolver] Add support for inline enum refactoring

### PR DESCRIPTION
**Pull request by wing328**
Add support for inline enum refactoring, which means inline enum schemas will be refactored separately and use $ref to reference them instead.

To close https://github.com/OpenAPITools/openapi-generator/issues/15972

To enable it, add `--inline-schema-name-defaults RESOLVE_INLINE_ENUMS=true` to CLI for example.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

FYI @OpenAPITools/generator-core-team

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java [hunk 1]
```diff
@@ -91,6 +91,16 @@ public void setInlineSchemaNameDefaults(Map inlineSchemaNameDefaults) {
         } else {
             // not set so default to null;
         }
+
+        if (this.inlineSchemaNameDefaults.containsKey("RESOLVE_INLINE_ENUMS")) {
+            if (Boolean.valueOf(this.inlineSchemaNameDefaults.get("RESOLVE_INLINE_ENUMS"))) {
+                this.resolveInlineEnums = true;
```
Comment (devhl-labs) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java:97]: You can remove the if statement
`this.resolveInlineEnums = Boolean.valueOf(this.inlineSchemaNameDefaults.get("RESOLVE_INLINE_ENUMS"))`

File: docs/customization.md [hunk 2]
```diff
@@ -455,6 +455,7 @@ Note: Only arrayItemSuffix, mapItemSuffix are supported at the moment.
 There are 2 special values:
 - `SKIP_SCHEMA_REUSE=true` is a special value to skip reusing inline schemas.
 - `REFACTOR_ALLOF_INLINE_SCHEMAS=true` will restore the 6.x (or below) behaviour to refactor allOf inline schemas into $ref. (v7.0.0 will skip the refactoring of these allOf inline schmeas by default)
+- `RESOLVE_INLINE_ENUMS=true` will refactor inline enum defintiions into $ref
```
Comment (devhl-labs) [inline docs/customization.md:458]: typo in definitions

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java]: fixed

File: docs/customization.md (hunk 2 above)
Comment (wing328) [reply docs/customization.md]: fixed

**Comment by gramar** [comment]
How can I set this with the maven plugin? 
Maybe like this?
                            <inlineSchemaOptions>
                                <resolveInlineEnums>true</resolveInlineEnums>
                            </inlineSchemaOptions>

**Comment by gramar** [comment]
I have found the correct syntax. Here it is:
`<inlineSchemaOptions>RESOLVE_INLINE_ENUMS=true</inlineSchemaOptions>`

**Pull request merged**